### PR TITLE
chore: various petrinet renderer tweaks

### PR DESCRIPTION
**Pull request by mwdchang**
### Summary
This makes a handful of tweaks and address some of the initial feedback to make the PetriNet renderer more usable.

- Add explicit state to turn edit on/off, change background colour to indicate editing state
- Remove a-star routing for now, this makes things too swiggly
- Pass in sizing configurations for petri-parsing to IGraph, this reduced hardwired numbers downstream
- Fix a few selection state issues (selected vs not-selected)
- Add a first pass of creating nodes though a context menu
- Tweak the edges and marker-ends to make the arrow a bit more visible

image

Note some of these changes do not exactly align with the design, we will get there in several iterations.

### Test
In the model tab, click "edit model" to enter edit mode (background changes to yellow), here you can
- Select a node and create new edges
- Select a node and delete (backsapce/delete)
- Right click to create new node, currently you cannot modify the name

Once done click save. The model editor will be in readonly and none of the above actions will be available.

### TODO in next PR
- Actually saving the edited ACSet back into database
- Edit node names

### CODE REVIEW

File: packages/client/hmi-client/src/components/models/Model.vue [hunk 1]
```diff
@@ -88,8 +89,9 @@
 </template>
 
 <script setup lang="ts">
+import _, { isEmpty } from 'lodash';
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/models/Model.vue:92]:
```suggestion
import { isEmpty, remove } from 'lodash';
```

**Comment by Tom-Szendrey** [comment]
I am right clicking in the model window to create a new node but nothing is happening. Do i need to select anything to do this?

**Comment by Tom-Szendrey** [comment]
> 

I can add S and T by write clicking when I am not in edit mode. When I am in edit mode I cannot. Is this intentional?

File: packages/client/hmi-client/src/components/models/Model.vue [hunk 2]
```diff
@@ -170,26 +174,88 @@ watch(
 );
 
 const graphElement = ref<HTMLDivElement | null>(null);
+
+let renderer: PetrinetRenderer | null = null;
+let eventX = 0;
+let eventY = 0;
+
+const editorKeyHandler = (event: KeyboardEvent) => {
```
[the remark this answers is not in the document]
Comment (Tom-Szendrey) [reply packages/client/hmi-client/src/components/models/Model.vue:182]: Good enough for me!

**Review by Tom-Szendrey — APPROVED** [review]

**Pull request merged**